File: front-end/src/app/medical-worker/medical-worker-make-an-appointment/endpoints/gender-getall.endpoint.ts
import {Injectable} from "@angular/core";
import {MyBaseEndpoint} from "../../../MyBaseEndpoint";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {MyConfig} from "../../../My-Config";

@Injectable({providedIn: 'root'})
export class GenderGetAllEndpoint  implements  MyBaseEndpoint<void, GenderGetAllResponse>{
  constructor(public httpClient:HttpClient) {

  }

  obradi(request: void): Observable<GenderGetAllResponse> {
    let url = MyConfig.server_address + '/api/GenderGetAllEndpoint';

    return this.httpClient.get<GenderGetAllResponse>(url);
  }
}

export interface GenderGetAllResponse {
  genders: GenderGetAllResponseGender[]
}

export interface GenderGetAllResponseGender {
  genderID: number;
  genderDescription: string;
}

File: VitaCare_API/VitaCare_API/Endpoints/MedicalWorker/GetAllGenders/GenderGetAllEndpoint.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Storage.Json;
using VitaCare_API.Data;
using VitaCare_API.Endpoints.MedicalWorker.GetAllExaminations;
using VitaCare_API.Helpers;

namespace VitaCare_API.Endpoints.MedicalWorker.GetAllGenders
{
    [Route("api/[controller]")]
    [ApiController]
    public class GenderGetAllEndpoint : MyBaseEndpoint<NoRequest, GenderGetAllResponse>
    {
        private readonly ApplicationDbContext _applicationDbContext;

        public GenderGetAllEndpoint(ApplicationDbContext applicationDbContext)
        {
            _applicationDbContext = applicationDbContext;
        }
        [HttpGet]
        public override async Task<GenderGetAllResponse> Obradi([FromQuery]NoRequest request)
        {
            var data = await _applicationDbContext.Gender.Select(x => new GenderGetAllResponseGender
            {
                genderID = x.GenderID,
                genderDescription = x.GenderDescription
            }).ToListAsync();

            return new GenderGetAllResponse
            {
                Genders = data
            };
        }
    }
}
